projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -21,12 +21,17 @@ import {
   isNullOrUndefined,
   ControlsType,
   ArrayPropertyKey,
+  TypedAbstractControl,
+  TypedFormGroup,
 } from './ngx-sub-form-utils';
-import { FormGroupOptions, NgxFormWithArrayControls, OnFormUpdate, TypedFormGroup } from './ngx-sub-form.types';
+import { FormGroupOptions, NgxFormWithArrayControls, OnFormUpdate } from './ngx-sub-form.types';
 
-type MapControlFunction<FormInterface, MapValue> = (ctrl: AbstractControl, key: keyof FormInterface) => MapValue;
+type MapControlFunction<FormInterface, MapValue> = (
+  ctrl: TypedAbstractControl<any>,
```

Comment:
Why does the above have the generic set to `any`?

Couldn't it be:

```suggestion
  ctrl: TypedAbstractControl<MapValue>,
```

(as `MapValue` is defaulting to `FormInterface` if not passed)?

Reply:
I did not touch that part. Tbh I'm not 100% clear what is going on here. If you say I can type I'll add it to the commit but I don't know how to test if this breaks something beyond running CI etc.

Reply:
see comments here: https://github.com/cloudnc/ngx-sub-form/pull/139#discussion_r383039437